onnx/defs/math/defs.cc

```diff
@@ -1085,7 +1085,50 @@ ONNX_OPERATOR_SET_SCHEMA(
         .TypeConstraint(
             "T",
             OpSchema::all_tensor_types(),
-            "Constrain input and output types to all tensors."));
+            "Constrain input and output types to all tensors.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          // Type inference
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
+          // Shape Inference if 2nd input data (the target shape) is available
+          const TensorProto* target_shape_initializer = ctx.getInputData(1);
+          if (!target_shape_initializer) {
+            return;
+          }
+          // The targetShape vector represents the specified shape for output.
+          std::vector<int64_t> target_shape;
+          if (target_shape_initializer->has_raw_data()) {
+            const std::string& bytes = target_shape_initializer->raw_data();
+            target_shape.insert(
+                target_shape.end(),
+                reinterpret_cast<const int64_t*>(bytes.c_str()),
+                reinterpret_cast<const int64_t*>(bytes.c_str() + bytes.size()));
+          } else {
+            const auto& data = target_shape_initializer->int64_data();
+            target_shape.insert(target_shape.end(), data.begin(), data.end());
+          }
+          auto& input_shape = getInputShape(ctx, 0);
+          auto* output_shape =
+              ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
+          for (int i = 0; i < input_shape.dim_size(); ++i) {
+            auto* new_dim = output_shape->add_dim();
+            int64_t target_dim = static_cast<size_t>(i) < target_shape.size() ? target_shape[i] : 1;
```

Comment:
Dimensions are right aligned. Please check the case here: https://github.com/onnx/onnx/blob/master/docs/Operators.md#expand

Reply:
Oops, sorry for the incorrect understanding. Done